### PR TITLE
Add NSQ driver and update vendor

### PR DESCRIPTION
**Pull request by soverdrive**
NSQ driver is added. The comunication with NSQ will be done by using
NSQD, not NSQLookupd. All topics and channels will be created with
`#ephemeral` suffix.

Updating the vendor to include Go NSQ driver and missing packages from
`tomato` master branch.

This should help resolving issue #10

### CODE REVIEW

File: resource/queue/nsq/nsq.go [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+		return nil, err
+	}
+
+	return cNsq, nil
+}
+
+// to listen to target, all messages retrieved should be stored temporarily
+func (nm *nsqManager) Listen(target string) error {
+	nm.Lock()
+	cNsqCached, exist := nm.topicReadiness[target]
```
Comment (alileza) [inline resource/queue/nsq/nsq.go]: might be worth to create function for this like

```go
func (nm *nsqManager) getTopic(target string) {
nm.Lock()
defer nm.Unlock()

return nm.topicReadiness[target]
}
```

**Comment by alileza** [comment]
can we separate commit for vendor and resource?

File: resource/queue/nsq/nsq.go [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+	sync.Mutex
+	nsqd           string
+	producer       *nsq.Producer
+	topicReadiness map[string]*nsq.Consumer
+	data           map[string][][]byte
+	waitDuration   time.Duration
+}
+
+func Open(params map[string]string) (*nsqManager, error) {
+	nsqdAddress, ok := params["nsqd"]
```
[the remark this answers is not in the document]
Comment (soverdrive) [reply resource/queue/nsq/nsq.go]: agreed and the documentation should mention that tomato need `NSQD address`. I'll change it

**Comment by soverdrive** [comment]
sure, I will split the commit

File: resource/queue/nsq/nsq.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (alileza) [reply resource/queue/nsq/nsq.go]: and just for your information, documentation will be auto-generated based on https://github.com/alileza/tomato/blob/master/dictionary.yml

**Comment by alileza** [comment]
thank you @soverdrive 🙏

**Pull request merged**